### PR TITLE
issue #646

### PR DESCRIPTION
**Pull request by mikey75**
Override button

### CODE REVIEW

File: bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/dialog/Modal.java [hunk 1]
```diff
@@ -130,8 +135,24 @@ protected void populateItem(ListItem<Component> item) {
...
+     * 
+     * @param id The component id
+     * @return the close button
+     */
+    
+    protected Component createHeaderCloseButton(String id) {
+
+    	BootstrapButton btn = new BootstrapButton(id, Buttons.Type.Link);
+    	btn.setLabel(new  Model<String>("\u00D7"));
+    	btn.add(new AttributeModifier("class","close"));
```
Comment (martin-g) [inline bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/dialog/Modal.java]: There is no need of this AttributeModifier. The class is already in the markup. Same for the label model.

Comment (mikey75) [inline bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/dialog/Modal.java]: I wanted to retain the original markup. The button alone will change the markup. There's also css at play. If you don't override the button the modal will have no correct markup. I 'll try to post a quickstart when i come home.
cheers

Comment (mikey75) [inline bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/dialog/Modal.java]: And, additionaly - actually - I think the better way would be to make header a container that would take components - same as  footer gets buttons etc. Suppose the getHeader factory method was meant for that.

Comment (martin-g) [inline bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/dialog/Modal.java]: I'll take a look and test it tonight after work. No need of a quickstart. I can use the samples application.

Comment (mikey75) [inline bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/dialog/Modal.java]: Any feedback?

**Comment by martin-g** [comment]
I've simplified your version to use `WebMarkupContainer` instead of BootstrapButton and then there is no need to set the label and the CSS class in the Java code.

**Comment by martin-g** [comment]
Thanks for the contribution!

**Comment by mikey75** [comment]
Thank you!

**Pull request merged**